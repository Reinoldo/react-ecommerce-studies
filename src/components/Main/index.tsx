import * as S from './styles'

const Main = ({
  title = 'React avancado',
  description = 'Bem vindo ao curso de react, typescript e next js'
}) => (
  <S.Wrapper>
    <S.Logo
      src="img/logo.svg"
      alt="Um atomo em formato de hexagono com o texto React avanÃ§ado do lado"
    />
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
    <S.Illustration
      src="img/hero-illustration.svg"
      alt="um desenvolvedor olhando para 3 monitores com codigo"
    />
  </S.Wrapper>
)
export default Main
