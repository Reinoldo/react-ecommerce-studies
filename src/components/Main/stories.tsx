import React from 'react'
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0'

import Main from '.'

export default {
  title: 'Example/Main',
  component: Main
} as Meta

export const Basic: Story = (args) => <Main {...args} />
