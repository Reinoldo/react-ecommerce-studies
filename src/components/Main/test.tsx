import { render } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the colours correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
    expect(container.firstChild).toMatchSnapshot()
  })
})
