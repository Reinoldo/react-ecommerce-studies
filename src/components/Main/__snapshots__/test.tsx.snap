// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the colours correctly 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  margin-bottom: 2rem;
  width: 25rem;
}

.c2 {
  font-weight: 400;
  font-size: 2rem;
}

.c3 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Um atomo em formato de hexagono com o texto React avanÃ§ado do lado"
    class="c1"
    src="img/logo.svg"
  />
  <h1
    class="styles__Title-sc-1cs6c8-2"
  >
    React avancado
  </h1>
  <h2
    class="c2"
  >
    Bem vindo ao curso de react, typescript e next js
  </h2>
  <img
    alt="um desenvolvedor olhando para 3 monitores com codigo"
    class="c3"
    src="img/hero-illustration.svg"
  />
</main>
`
